### PR TITLE
declare Cython dependency

### LINKED ISSUE #25: missing: requirements.txt
when trying to install this package from source, we hit the issue where it doesn't declare its own dependencies thru requirements.txt or other means and as such, fails to compile because it's missing `Cython`

```
Collecting libzfs from git+https://github.com/freenas/py-libzfs.git#egg=libzfs (from -r requirements.txt (line 10))
  Cloning https://github.com/freenas/py-libzfs.git to /tmp/pip-build-pllsc0sm/libzfs
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-pllsc0sm/libzfs/setup.py", line 27, in <module>
        import Cython.Compiler.Options
    ModuleNotFoundError: No module named 'Cython'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-pllsc0sm/libzfs/
```

(please do not be too concerned with freenas in the url, as that part has definitely not changed in their repo ;)

### PR DESCRIPTION
**Pull request by igalic**
* add setup_requires.txt with Cython
* add setup_requires with Cython

for the latter, use setuputils>=18.0 because it can handle Cython extensions
We do this at the loss of Compiler.Options.annotate

Regardless, none of this fixes  https://github.com/jceel/py-libzfs/issues/25
It's practically impossible to bootstrap setup(), since even when it
installs requirements, it does not reload *itself*.

in testing this fix, we also discovered that .gitignore was missing.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -24,14 +24,15 @@
...
-from distutils.core import setup
-from Cython.Distutils.extension import Extension
-from Cython.Distutils import build_ext
+from setuptools import setup
 
+try:
+    from Cython.Build import build_ext
+    from Cython.Distutils.extension import Extension
+except ImportError:
+    raise "This package requires Cython to build properly. Please install it first."
```
Comment (william-gr) [inline setup.py]: You cannot raise a string

File: setup.py [hunk 2]
```diff
@@ -24,14 +24,15 @@
...
-from distutils.core import setup
-from Cython.Distutils.extension import Extension
-from Cython.Distutils import build_ext
+from setuptools import setup
 
+try:
+    from Cython.Build import build_ext
+    from Cython.Distutils.extension import Extension
+except ImportError:
+    raise ImportError("This package requires Cython to build properly. Please install it first.")
```
Comment (igalic) [inline setup.py:35]: @william-gr how about an ImportError then?

**Comment by skarekrow** [comment]
LGTM @william-gr

File: setup.py (hunk 2 above)
Comment (william-gr) [reply setup.py:35]: Better, thats at least logically correct ;)

**Review by william-gr — APPROVED** [review]

**Pull request merged**